### PR TITLE
virt-controller: fix double `time.Second` multiplication in rate limiter

### PR DESCRIPTION
**Pull request by jean-edouard**
**What this PR does / why we need it**:
While working on metrics, I found this double multiplication which transforms a 5 second value into a ~158 year one, which seems wrong.
This PR fixes that in its first commit, and provides some nit-picky code cleanup in its second commit.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
The `defaultThrottleInterval` const is still used for 3 completely unrelated things, none of which seem to match its comment.
This code probably needs further cleanup.

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: The [Boy Scout Rule](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html) was considered and applied if needed
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to the [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jean-edouard** [comment]
/cc @davidvossel 
This touches code from that PR: https://github.com/kubevirt/kubevirt/pull/5342
Could you please shed some light on why we went from a base delay of 5ms in the default limiter to 5s in the custom one? Seems like a big jump!
Similarly, the bucket rate limiter went from 10 qps to... 0.2 qps? (1 every 5 seconds)
Thank you!

**Comment by enp0s3** [comment]
/approve
@jean-edouard Thank you!

**Comment by enp0s3** [comment]
@jean-edouard Thank you!
/approve

**Comment by fossedihelm** [comment]
Looks good!
/lgtm
/retest-required

File: pkg/virt-controller/watch/workload-updater/workload-updater.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ var (
 const periodicReEnqueueIntervalSeconds = 30
 
 // ensures we don't execute more than once every 5 seconds
-const defaultThrottleIntervalSeconds = 5 * time.Second
+const defaultThrottleInterval = 5 * time.Second
```
Comment (xpivarc) [inline pkg/virt-controller/watch/workload-updater/workload-updater.go:61]: Why not keep "Seconds"?

File: pkg/virt-controller/watch/workload-updater/workload-updater.go [hunk 2]
```diff
@@ -102,8 +103,8 @@ func NewWorkloadUpdateController(
 ) (*WorkloadUpdateController, error) {
 
 	rl := workqueue.NewMaxOfRateLimiter(
-		workqueue.NewItemExponentialFailureRateLimiter(time.Duration(defaultThrottleIntervalSeconds)*time.Second, 300*time.Second),
```
Comment (xpivarc) [inline pkg/virt-controller/watch/workload-updater/workload-updater.go:105]: FYI, This basically means it always returned a max, of 300 Seconds.

Comment (jean-edouard) [reply pkg/virt-controller/watch/workload-updater/workload-updater.go:105]: Right, but I'm not sure that reasoning applies to the bucket one just below, and `NewMaxOfRateLimiter()` returns the longest one.

**Pull request merged**